posthog/hogql/database/schema/persons.py

```diff
@@ -46,72 +44,73 @@ def select_from_persons_table(
+    # We only do this if there are where clauses, _and_ WhereClauseExtractor can extract them
+    if node.where:
+        inner_select = cast(
             ast.SelectQuery,
             parse_select(
                 """
-            SELECT id FROM raw_persons WHERE (id, version) IN (
-               SELECT id, max(version) as version
-               FROM raw_persons
-               GROUP BY id
-               HAVING equals(argMax(raw_persons.is_deleted, raw_persons.version), 0)
-               AND argMax(raw_persons.created_at, raw_persons.version) < now() + interval 1 day
-            )
+            SELECT id
+            FROM raw_persons
+            WHERE
+                -- Much faster to pre-select out any deleted persons than doing it in aggregation
+                -- This is correct because there are no instances where we'd un-delete a person (ie there are no cases where one version has is_deleted=1 and a later version has is_deleted = 0)
+                id NOT IN (select id from raw_persons where is_deleted = 1)
             """
             ),
         )
-        select.settings = HogQLQuerySettings(optimize_aggregation_in_order=True)
-        if filter is not None:
-            cast(ast.SelectQuery, cast(ast.CompareOperation, select.where).right).where = filter
-
-        for field_name, field_chain in join_or_table.fields_accessed.items():
-            # We need to always select the 'id' field for the join constraint. The field name here is likely to
-            # be "persons__id" if anything, but just in case, let's avoid duplicates.
-            if field_name != "id":
-                select.select.append(
-                    ast.Alias(
-                        alias=field_name,
-                        expr=ast.Field(chain=field_chain),
-                    )
-                )
-    else:
-        select = argmax_select(
-            table_name="raw_persons",
-            select_fields=join_or_table.fields_accessed,
-            group_fields=["id"],
-            argmax_field="version",
-            deleted_field="is_deleted",
-            timestamp_field_to_clamp="created_at",
-        )
-        select.settings = HogQLQuerySettings(optimize_aggregation_in_order=True)
-        if filter is not None:
-            if select.where:
-                select.where = And(exprs=[select.where, filter])
-            else:
-                select.where = filter
-
-    if context.modifiers.optimizeJoinedFilters:
-        extractor = WhereClauseExtractor(context)
+        extractor = WhereClauseExtractor(context, is_join=is_join)
         extractor.add_local_tables(join_or_table)
         where = extractor.get_inner_where(node)
-        if where and select.where:
-            select.where = And(exprs=[select.where, where])
-        elif where:
-            select.where = where
+
+        if where and inner_select.where:
```

Comment:
It might be empty if we didn't manage to extract any clauses, in which case we don't want to use this optimization (as you'd just be doing a subquery with all persons).